### PR TITLE
File assert the things

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**

### CODE REVIEW

File: stable/overlays/openstack-base-virt-overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+applications:
+  ceph-osd:
+    to: []
+    storage:
+      osd-devices: 20GB
+  neutron-gateway:
+    options:
+      bridge-mappings: physnet1:br-ex
+      data-port:
+      openstack-origin: cloud:bionic-rocky
```
Comment (gnuoy) [inline stable/overlays/openstack-base-virt-overlay.yaml]: Why is openstack-origin set here?

File: stable/shared/file-assertions.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
+ceph-mon:
+  files:
+    - path: /etc/ceph/ceph.conf
+      owner: root
+      group: root
+      mode: '777'
```
Comment (gnuoy) [inline stable/shared/file-assertions.yaml]: Why are we checking that ceph.conf is executable ?

File: stable/shared/file-assertions.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    - path: /var/lib/charm/ceph-mon/ceph.conf # /etc/alternatives/ceph.conf -> /etc/ceph/ceph.conf destination
+      owner: root
+      group: root
+      mode: '644'
+    - path: /etc/ceph/ceph.client.admin.keyring
```
Comment (gnuoy) [inline stable/shared/file-assertions.yaml]: Shouldn't we be looking for a mode of 600?

File: stable/shared/file-assertions.yaml (hunk 2 above)
Comment (gnuoy) [reply stable/shared/file-assertions.yaml]: oh hang on , is this a symlink?

Comment (ChrisMacNaughton) [reply stable/shared/file-assertions.yaml]: that's an `install-alternative` 0 the /var/lib/juju/$whatever` path is to the real file

File: stable/shared/file-assertions.yaml (hunk 3 above)
Comment (ChrisMacNaughton) [reply stable/shared/file-assertions.yaml]: the default parameters in the checks are root:root, mode 600, so an undefined one will get 600 :)

File: stable/overlays/openstack-base-virt-overlay.yaml (hunk 1 above)
Comment (ChrisMacNaughton) [reply stable/overlays/openstack-base-virt-overlay.yaml]: whoops :) fix incoming

**Pull request merged**